File: backend/app/graphql/types/comparison.py
from __future__ import annotations

import strawberry

from app.graphql.types.car import CarType


@strawberry.type
class ComparisonRowType:
    key: str
    label: str
    winner_variant_id: str | None
    values: list[str]


@strawberry.type
class ComparisonType:
    cars: list[CarType]
    winner_per_row: list[ComparisonRowType]
    verdict: str
    buyer_type_match: list[str]

File: frontend/src/graphql/queries/comparison.ts
import { gql } from "@apollo/client";

import { CAR_FIELDS } from "../fragments/car";

export const COMPARE_QUERY = gql`
  ${CAR_FIELDS}
  query Compare($variantIds: [ID!]!) {
    compare(variantIds: $variantIds) {
      cars {
        ...CarFields
      }
      winnerPerRow {
        key
        label
        winnerVariantId
        values
      }
      verdict
      buyerTypeMatch
    }
  }
`;
